includes/admin/class-wc-rest-stripe-settings-controller.php

```diff
@@ -388,6 +414,29 @@ private function update_is_test_mode_enabled( WP_REST_Request $request ) {
 		$this->gateway->update_option( 'testmode', $is_test_mode_enabled ? 'yes' : 'no' );
 	}
 
+	/**
+	 * Get the merchant payment method configuration in Stripe.
+	 *
+	 * @return array|null
+	 */
+	private function get_merchant_payment_method_configuration() {
+		$payment_method_configurations = $this->stripe_api->get_payment_method_configurations() && property_exists( $this->stripe_api->get_payment_method_configurations(), 'data' )
```

Comment:
In addition to what Anne mentioned below, calling the Stripe API every time the settings page will impact page load performance.

Is there a way to avoid that? 

One approach is to tie it up with the existing "Refresh account details" button and avoid automatic fetches.  
If configuration updates result in a webhook event, another approach might be to listen to those for updates.

Reply:
On a separate note, we'd want to add a log if the call to Stripe API fails for some reason.

```
if ( is_wp_error( $response ) ) {
    WC_Stripe_Logger::log( 'Failed to fetch payment method configurations from Stripe API.' );
    return null;
}
```

Reply:
>One approach is to tie it up with the existing "Refresh account details" button and avoid automatic fetches.
If configuration updates result in a webhook event, another approach might be to listen to those for updates.

This could work, but this would change how the feature works and prevent users who update settings from the Stripe dashboard from seeing them immediately in admin settings. @diegocurbelo, would this work for us?

Reply:
@diegocurbelo I think that makes sense.  One thing to keep in mind is that we will also need to fetch the stripe payment configuration on every checkout page because we need to check for enabled payment methods there too.

Is that reasonable or should we implement non-admin only caching?

Reply:
Non-admin only caching is a great idea 👍 … the checkout can potentially generate a lot of requests to the configurations api. And we can update that cache when we fetch new data in the admin pages.